api/controller/dbController.js

```diff
@@ -0,0 +1,94 @@
+import { query } from '../../db/db'
+import { 
+  artistQueries, setlistQueries, songQueries, setlistSongQueries
+  } from '../../db/queries'
+
+  export class dbController {
+    artistId;
+    concertId;
+    songId;
+    constructor(apiResult, artist, song, artistId, concertId, songId) {
+      this.apiResult = apiResult;
+      this.artist = artist;
+      this.song = song;
+      this.externalId = apiResult.setlist[0].artist.mbid;
+      this.artistId = artistId;
+      this.concertId = concertId;
+      this.songId = songId;
+    }
+
+    addSong = async (song) => {
+      const addResult = await query(songQueries.addSong,[song,this.artistId]);
+      this.songId = addResult[0].id;
+    }
+    
+    getSong = async (song) => {
+      const getResult = await query(songQueries.getSong,[song,this.artistId]);
+      const songInDb = getResult.length > 0;
+      if (songInDb) {
+        this.songId = getResult[0].id;
+      } else {
+        this.songId = null;
+      }
+    }
+
+    addSetlistSong = async () => {
+      await query(setlistSongQueries.addSetlistSong,[this.concertId,this.songId])
+    }
+
+    addSetlist = async (externalId,updatedAt,eventDate) => {
+      const result = await query(setlistQueries.addSetlist,[externalId,this.artistId,updatedAt,eventDate]);
+      return result[0].id;
+    }
+  
+    addedArtist = async () => {
+      const result = await query(artistQueries.addArtist,[this.externalId,this.artist]);
+      this.artistId = result[0].id;
+    }
+  
+    getArtist = async () => {
+      const result = await query(artistQueries.getArtist,[this.artist]);
+      this.artistId = result.length > [0] ? result[0].id : 0;
+      return this.artistId;
+    }
+
+    getDates = async () => {
+      const returnedValue = {};
+      const result = await query(setlistSongQueries.getSetlistSong,[this.song,this.artistId])
+      const compiledDates = this.compileDates(result);
+      returnedValue[this.song] = compiledDates;
+      return returnedValue;
+    }
+    
+    compileSongs = (sets) => {
+      const songsFromThisConcert = [];
+      sets.forEach(set => set.song.forEach(song => songsFromThisConcert.push(song.name.toLowerCase())));
+      return songsFromThisConcert;
+    }
```

Comment:
this is a good idea to make the downcase Names function. However,  this function is more about taking values from different levels of nested objects and putting them in one array. When I get the song data from the API for an individual concert it's in the form
{set:[
          {
            song:[
                       {
                        name: foo
                      },
                      {
                      name: bar
                      }
                   ]
              },
            {
              song:[
                       {
                          name: cool
                      },
                      {
                          name: stuff
                      }
                   ]
              },
         ]
}
and the function returns [foo,bar,cool,stuff]. 

that being said it's still clearly too vague. maybe 'deriveSongsFromSets' would be a better name? The lowercase function does sound like a good idea to make the code cleaner since rn the function in doing 2 things at once. I'll make the lowercase part a separate function and call it after the songs are all in one array.

Reply:
Oohhh I missed a level in there! Didn't realize that this is getting the songs from the sets as well!